testkit/src/checkpoint_db.rs

```diff
@@ -73,31 +74,39 @@ pub struct CheckpointDbHandler<T> {
 }
 
 impl<T: Database> CheckpointDbHandler<T> {
-    /// Rolls back this database by undoing the latest `count` `merge()` operations.
+    /// Sets a checkpoint for a future [`rollback`](#method.rollback).
+    pub fn checkpoint(&self) {
+        self.inner
+            .write()
+            .expect("Cannot lock checkpointDb for checkpoint")
+            .checkpoint();
+    }
+
+    /// Rolls back this database to the latest checkpoint
+    /// set with [`checkpoint`](#method.checkpoint).
     ///
     /// # Panics
     ///
-    /// - Panics if more operations are attempted to be reverted than the number of operations
-    ///   in the DB journal.
-    pub fn rollback(&self, count: usize) {
+    /// - Panics if there are no available checkpoints.
+    pub fn rollback(&self) {
         self.inner
             .write()
             .expect("Cannot lock CheckpointDb for rollback")
-            .rollback(count);
+            .rollback();
     }
 }
 
 #[derive(Debug)]
 struct CheckpointDbInner<T> {
     db: T,
-    journal: Vec<Patch>,
+    backup_stack: Vec<VecDeque<Patch>>,
```

Comment:
Nitpick: I don't particularly understand the reasoning behind renaming, but I'm OK with the new name.

Reply:
I find `journal` name very ambiguous. `backup_stack` on the other hand, gives an idea on two things:
* This `Vec` is used like a stack;
* It contains "backup" patches as opposed to "change" patches.

Reply:
I've taken the name from [DB terminology](https://en.wikipedia.org/wiki/Transaction_log), but I see your point.

Reply:
From the [article](https://en.wikipedia.org/wiki/Transaction_log):
> Physically, a log is a file listing changes to the database, stored in a stable storage format.

This contradicts with our case because we don't store actual changes to the database but rather reversed changes.